client/tests/integration/asset.rs

```diff
@@ -160,7 +160,7 @@ fn client_add_asset_with_decimal_should_increase_asset_amount() -> Result<()> {
         )),
     );
     let instructions: Vec<Instruction> = vec![create_asset.into(), mint.into()];
-    let tx = test_client.build_transaction(instructions.into(), metadata)?;
+    let tx = test_client.build_transaction(instructions, metadata)?;
```

Comment:
this sort of generic API makes sense for client facing code. For internal code I would rather we didn't have generic apis because it might bloat the binary unnecessarily due to monomorphisation. We should resolver types before calling the function for iroha_core whenever possible